decoder.go

```diff
@@ -34,28 +34,29 @@ func NewDecoder(r io.Reader) *Decoder {
 // Graceful disconnects (between events) are indicated by an io.EOF error.
 // Any error occuring mid-event is considered non-graceful and will
 // show up as some other error (most likely io.ErrUnexpectedEOF).
-func (dec *Decoder) Decode() (Event, error) {
+func (dec *Decoder) Decode() (Event, *string, error) {
```

Comment:
Now that there are 3 return values, it would be useful in the comments to describe what the *string is that we're returning.